src/System.Diagnostics.Process/tests/ProcessTests.Unix.cs

```diff
@@ -783,6 +783,61 @@ public void TestProcessRecycledPid()
             Assert.True(foundRecycled);
         }
 
+        [Theory]
+        [InlineData(true)]
+        [InlineData(false)]
+        public async Task Kill_ExitedNonChildProcess_DoesNotThrow(bool killTree)
+        {
+            // In this test, we kill a process in a way the Process instance
+            // is not aware the process has terminated when we invoke Process.Kill.
+
+            using (Process nonChildProcess = CreateNonChildProcess())
+            {
+                // Kill the process.
+                int rv = kill(nonChildProcess.Id, SIGKILL);
+                Assert.Equal(0, rv);
+
+                // Wait until the process is reaped.
+                for (int i = 0; i < 100 && rv == 0; i++)
+                {
+                    rv = kill(nonChildProcess.Id, 0);
+                    if (rv == 0)
+                    {
+                        // process still exists, wait some time.
+                        await Task.Delay(100);
+                    }
+                }
```

Comment:
Loops like this very frequently come back to bite us, as at some point this is almost certain to fail in CI, with the reaping taking more than 10s at some point.  There must be a more deterministic way to do this, no?

Reply:
The corefx implementation is similar to what we are doing here (without a timeout). We can increase the timeout if 10s is too small.

Reply:
I use a rule of thumb of a timeout that's maybe 10 times larger than it ever takes on a dev machine (up to maybe 30 sec) is necessary to make it solid on a CI machine.